src/ducks/Watchlists/Actions/Edit/EditForm/Assets.js

```diff
@@ -0,0 +1,185 @@
+  const { isNightMode } = useTheme()
+  const { data } = useAllProjects(filter.toLowerCase())
+  const assets = useMemo(
+    () => data.filter(item => !watchListIDs.includes(item.id)),
+    [data, watchListIDs]
+  )
+  const [showItems, setShowItems] = useState(false)
+
+  useEffect(() => {
+    const showTimeout = setTimeout(() => setShowItems(true), 300)
+    return () => clearTimeout(showTimeout)
+  }, [])
+
+  useOnClickOutside(ref, () => {
+    setFilter('')
+    setIsSearchMode(false)
+  })
+
+  useEffect(() => {
+    let _items = watchListItems
+    if (filter && filter.length > 0) {
+      const _filter = filter.toLowerCase()
+      const filterHelper = item =>
+        item.name.toLowerCase().includes(_filter) ||
+        item.ticker.toLowerCase().includes(_filter)
+      _items = _items.filter(filterHelper)
+    }
+    setItems(_items)
+  }, [filter, watchListItems])
+
+  const checkboxClickHandler = (item, newValue) =>
+    setCheckedItems(old => {
+      const items = [...old]
+      const index = items.findIndex(i => i.id === item.id)
+      if (index === -1 && newValue) {
+        items.push(item)
+      } else if (index > -1 && !newValue) {
+        items.splice(index, 1)
+      }
+      if (onChange) onChange(items)
+      return items
+    })
+
+  return (
+    <>
+      <Label accent='waterloo' className={styles.description__label}>
+        Assets
+      </Label>
+      <div
+        ref={ref}
+        className={cx(fieldStyles.selector, styles.selector)}
+        onClick={() => !isSearchMode && setIsSearchMode(true) && setFilter('')}
+      >
+        {!isSearchMode && (
+          <>
+            {checkedItems.length === 0 && (
+              <span className={cardStyles.ticker}>
+                Select asset for watchlist
+              </span>
+            )}
+            {checkedItems.length > 0 &&
+              checkedItems.slice(0, VIEW_ITEM_COUNT).map((item, index) => {
+                const hasDots =
+                  checkedItems.length > VIEW_ITEM_COUNT &&
+                  index === VIEW_ITEM_COUNT - 1
+                const hasComma =
+                  index < VIEW_ITEM_COUNT - 1 && index < checkedItems.length - 1
+                const seprator = hasComma ? ', ' : hasDots ? '...' : ''
+                return (
+                  <div
+                    className={cx(cardStyles.name, styles.mrhalf)}
+                    key={item.id}
+                  >
+                    {item.name}{' '}
+                    <span className={cardStyles.ticker}>{item.ticker}</span>{' '}
+                    {seprator}
+                  </div>
+                )
+              })}
+          </>
+        )}
+
+        {isSearchMode && (
+          <Input
+            autoFocus
+            maxLength='25'
+            autoComplete='off'
+            placeholder='Type to search'
+            className={styles.searchInput}
+            onChange={({ currentTarget: { value } }) => setFilter(value)}
+          />
+        )}
```

Comment:
To improve readability better to move the `LOC 81–106` to outside component (`Component1`). This will allow to conditionally render `isSearchMode ? <Input ... /> : <Component1 />`  in a single instruction